### ISSUE TITLE
Codable routes can be misleading

### ISSUE THREAD

**Issue by awadodeh** [IBM-Swift/Kitura#1473]
While developing routes using the codable routes feature in Kitura I found that it could introduce some readability issues that can be resolved if parameters are allowed to be added to the routes. 

This is a working example:

```swift
class ReviewController {
    private let router: Router
    private let repository: ReviewRepository
 
    init(_ router: Router, repository: ReviewRepository) {
        self.router = router
        self.repository = repository
        router.get("/reviews", handler: self.handleGetAllReviews)
        router.get("/reviews", handler: self.handleGetReviewById)
        router.post("/reviews", handler: self.handleAddReview)
        router.delete("/reviews", handler: self.deleteReviewById)
    }

    private func handleGetAllReviews(completion: @escaping ([Review]?, RequestError?) -> Void){
        Log.debug("Trying to get all reviews.")
        self.repository.findAll() { reviews, error in
            return completion(reviews, error as? RequestError)
        }
    }

    private func handleGetReviewById(id: String, completion: @escaping (Review?, RequestError?) -> Void) {
        Log.debug("Trying to find a review by id: \(id).")
        self.repository.findReview(reviewId: id) { reviews, error in
            return completion(reviews, error as? RequestError)
        }
    }

    private func handleAddReview(review: Review, completion: @escaping (Review?, RequestError?) -> Void) {
        Log.debug("Trying to add review.")
        self.repository.addReview(review: review) { review, error in
            return completion(review, error as? RequestError)
        }
    }

    private func deleteReviewById(id: String, completion: @escaping (RequestError?) -> Void) {
        Log.debug("Trying to add review.")
        self.repository.deleteReview(reviewId: id) { error in
            return completion(error as? RequestError)
        }
    }
}
```

But if I added params to the routes it does not detect them.

```swift
class ReviewController {
    private let router: Router
    private let repository: ReviewRepository
 
    init(_ router: Router, repository: ReviewRepository) {
        self.router = router
        self.repository = repository
        router.get("/reviews", handler: self.handleGetAllReviews)
        router.get("/reviews/:id", handler: self.handleGetReviewById)
        router.post("/reviews", handler: self.handleAddReview)
        router.delete("/reviews/:id", handler: self.deleteReviewById)
    }

    private func handleGetAllReviews(completion: @escaping ([Review]?, RequestError?) -> Void){
        Log.debug("Trying to get all reviews.")
        self.repository.findAll() { reviews, error in
            return completion(reviews, error as? RequestError)
        }
    }

    private func handleGetReviewById(id: String, completion: @escaping (Review?, RequestError?) -> Void) {
        Log.debug("Trying to find a review by id: \(id).")
        self.repository.findReview(reviewId: id) { reviews, error in
            return completion(reviews, error as? RequestError)
        }
    }

    private func handleAddReview(review: Review, completion: @escaping (Review?, RequestError?) -> Void) {
        Log.debug("Trying to add review.")
        self.repository.addReview(review: review) { review, error in
            return completion(review, error as? RequestError)
        }
    }

    private func deleteReviewById(id: String, completion: @escaping (RequestError?) -> Void) {
        Log.debug("Trying to add review.")
        self.repository.deleteReview(reviewId: id) { error in
            return completion(error as? RequestError)
        }
    }
}
```

**Comment by djones6** [comment]
This is a bug - the intention (as of https://github.com/IBM-Swift/Kitura/pull/1310 / Kitura 2.6) was that users should specify `:id` on routes that expect an identifier.  With logging enabled, you'll see a warning that `:id` was added implicitly if you do not specify it yourself.

However, it looks like the implementation was incorrect: although specifying `:id` silences the warning, Kitura then appends a second `:id` parameter and that second parameter is the one that is used.  I'll investigate a fix for this.

**Comment by djones6** [comment]
This has been fixed in Kitura `2.8.1`.  I'll back-port the fix to the 2.7 release in https://github.com/IBM-Swift/Kitura/pull/1485
